fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biomes/BiomeAssociate.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.biomes;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.biome.Biome;
+import net.minecraft.world.biome.layer.LayerRandomnessSource;
+
+public class BiomeAssociate
+{
+	private int weightSum = 0;
+	private List<Integer> biomes = new ArrayList<>();
```

Comment:
This should be an IntList from fastutils.

Reply:
Use multiset.

Reply:
@liach what is a multiset

Reply:
Also this class exists because it uses LayerRandomnessSource as the random generator, not java.util.Random

Reply:
Ooh... I didn't change to an intlist. Will do that

Reply:
Changed to intlist, as requested.

Reply:
Reverted because of NullPointerExceptions

Reply:
What NPEs?

Reply:
If you're getting NPEs then fix that or explain the need.  I agree using a boxed type here makes no sense.

Reply:
No longer an issue, this class is gone entirely :P